records/2025-02-19.authmethods.md

```diff
@@ -0,0 +1,174 @@
+• Consider multiple authentication options for user flexibility   
+• If choosing between email and SMS, email generally offers better security against targeted attacks    
+
+# Recommendations
+
+The options below are not independent of each other. We can implement multiple options going forward.
+My recommendation is to give users options for different 2FA types. This way, users can choose what is convenient
+for them while we maintain control over the types of auth flows (for security purposes).
+
+
+## 1. Remove SMS as a 2FA option  
+
+Currently, Notify requires a user's phone number in the signup flow. 
+1. We can remove SMS as an option going forward.  
+2. We will need a migration plan to move existing SMS 2FA users to another auth_type.  
+3. We would then disable 2FA verification by SMS for all Notify users.  
+
+Benefits of removing SMS-based 2FA:  
+• Eliminates vulnerabilities to SIM swapping attacks  
+• SMS is known to be the least secure form of 2FA
+• Improves overall security posture
+
+
+## 2. Mobile based Authenticator Apps (TOTP - Time based one time password)
+Authenticator apps are widely used by multiple clients and teams for signup flows. Most users will have encountered them while using everyday banking or mail applications.
+Authenticators provide one-time passwords generated by applications like Google Authenticator, Microsoft Authenticator, or Authy.
+Introducing an authenticator application will not solve the issue of users who do not have a cell phone provided by their department.
+This is just an additional 2FA method for clients with a department-provided cellphone.
+
+### Benefits:  
+• Works offline once configured  
+• Not vulnerable to SIM swapping  
+• No SMS costs  
+• Higher security than SMS  
+
+### Cons:  
+• Requires smartphone and app installation  
+• Device loss can lead to account recovery challenges  
+• Slightly higher friction during initial setup  
+
+### Implementation:    
+The FIDO2 library that Notify currently uses for implementing YubiKey can be extended to include Authenticator apps.
+The implementation would follow the details for YubiKey and store login events in the sessions table.
+
+## 3. [Magic Link ADR](https://github.com/cds-snc/notification-adr/pull/64/files)
+We can revive the work done in the above ADR for magic links. There is a stock implementation in Notify, we could use for Magic Links.
+Please see the above ADR for introducing magic links into our 2FA flow
+
+## 4. Default to using email for 2FA
+If we remove SMS as a 2FA method, we should default the 2FA to email authentication.  
+• Requires minimal changes to authentication framework as this is already implemented  
+• Verification logic remains largely unchanged from SMS implementation  
+
+# Decision
+
+We are not going to offer SMS 2FA auth going forward. There are two reasons for this:  
+1. SMS 2FA is not secure (as shown in recommendation 1)
+2. Clients without a phone should still be able to sign up for notify
```

Comment:
This is per my understanding the main issue that was raised to us, and that this ADR needs to solve as per the opening problem. But I don't see why the removal of SMS as a factor authentication is required. 

I am unsure but the only blocker there might be is that we make phone numbers a required part of our registration?  If we remove that requirement, users could then opt for SMS or email for their preferred 2nd factor.